corehq/apps/domain/forms.py

```diff
@@ -472,8 +480,26 @@ def clean_default_timezone(self):
         timezone_field.run_validators(data)
         return smart_str(data)
 
+    def clean_alias(self):
+        data = self.cleaned_data['alias'].strip().lower()
+        if data == "":
+            return data
+
+        if not re.match("^%s$" % new_domain_re, data):
+            raise forms.ValidationError('Only lowercase letters and numbers allowed. ' +
+                'Single hyphens may be used to separate words.')
```

Comment:
Also, python does automatic string concatenation inside parenthesis, so you can do:
```python
raise forms.ValidationError(_('Only lowercase letters and numbers allowed. '
                              'Single hyphens may be used to separate words.'))
```